string name
float64 x
float64 y
float64 w
---
bool success
string message
